front/src/applications/stdcm/views/StdcmRequestModal.tsx

```diff
@@ -48,71 +54,77 @@ export default function StdcmRequestModal(props) {
   // https://developer.mozilla.org/en-US/docs/Web/API/AbortController
   const controller = new AbortController();
 
-  const timetableId = osrdconf.timetableID;
+  const timetableId = osrdconf.timetableID?.toString();
 
   // Returns a promise that will be a fetch or an axios (through react-query)
   const stdcmRequest = async () => {
-    const params = formatStdcmConf(dispatch, setFailure, t, osrdconf);
-    return postStdcm(params).unwrap();
+    const params = formatStdcmConf(dispatch, setFailure, t, osrdconf as OsrdStdcmConfState);
+    return postStdcm(params as PostStdcmApiArg).unwrap();
```

Comment:
Could you move these 2 lines directly in the useEffect ? And also remove the cast of PostStdcmApiArg and send the request only if the payload is not undefined ?
```suggestion
    const payload = formatStdcmConf(dispatch, setFailure, t, osrdconf as OsrdStdcmConfState);
    return postStdcm(payload).unwrap();
```

Reply:
done